src/transformStateWithClones.js

```diff
@@ -6,8 +6,31 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
```

Comment:
Use destructuring for `action`
```suggestion
  actions.forEach((action) => {
    const { type, ...... } = action;
    
    switch (type) {
```